core/trino-main/src/test/java/io/trino/block/BenchmarkRowBlockBuilder.java

```diff
@@ -0,0 +1,106 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.block;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.spi.block.RowBlockBuilder;
+import io.trino.spi.block.SingleRowBlockWriter;
+import io.trino.spi.type.RowType;
+import io.trino.spi.type.Type;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.CompilerControl;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.profile.GCProfiler;
+
+import java.util.List;
+
+import static io.trino.jmh.Benchmarks.benchmark;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static java.util.concurrent.TimeUnit.MICROSECONDS;
+
+@SuppressWarnings("MethodMayBeStatic")
+@State(Scope.Thread)
+@OutputTimeUnit(MICROSECONDS)
+@Fork(1)
+@Warmup(iterations = 15)
+@Measurement(iterations = 15)
+@BenchmarkMode(Mode.AverageTime)
+public class BenchmarkRowBlockBuilder
+{
+    @Benchmark
+    @CompilerControl(CompilerControl.Mode.DONT_INLINE)
+    public void benchmarkBeginBlockEntryAlloc(BenchmarkData data, Blackhole blackhole)
+    {
+        for(int i = 0; i < data.rows; i++) {
+            SingleRowBlockWriter singleRowBlockWriter = data.blockBuilder.beginBlockEntryAlloc();
+            for(int fieldIndex = 0; fieldIndex < data.types.size(); fieldIndex++) {
+                singleRowBlockWriter.writeLong(10).closeEntry();
+            }
+            data.blockBuilder.closeEntry();
+            blackhole.consume(singleRowBlockWriter);
+        }
+    }
+
+    @Benchmark
+    @CompilerControl(CompilerControl.Mode.DONT_INLINE)
+    public void benchmarkBeginBlockEntryNoAlloc(BenchmarkData data, Blackhole blackhole)
+    {
+        for(int i = 0; i < data.rows; i++) {
+            SingleRowBlockWriter singleRowBlockWriter = data.blockBuilder.beginBlockEntry();
+            for(int fieldIndex = 0; fieldIndex < data.types.size(); fieldIndex++) {
+                singleRowBlockWriter.writeLong(10).closeEntry();
+            }
+            data.blockBuilder.closeEntry();
+            blackhole.consume(singleRowBlockWriter);
+        }
+    }
+
+    @State(Scope.Thread)
+    public static class BenchmarkData
+    {
+        private RowBlockBuilder blockBuilder;
+        private List<Type> types = ImmutableList.of(BIGINT, BIGINT);
```

Comment:
I would also parametrize length of `types` (e.g. `1, 2, 10`)

Reply:
ok